cli/Sources/TuistKit/Commands/Cache/Tuist+CacheProfileResolver.swift

```diff
@@ -0,0 +1,79 @@
+import Foundation
+import Logging
+import TuistCore
+
+enum CacheProfileError: LocalizedError, Equatable {
+    case profileNotFound(profile: String, available: [String])
+
+    var errorDescription: String? {
+        switch self {
+        case let .profileNotFound(profile, available):
+            let builtins = "only-external, all-possible, none"
```

Comment:
Let's make the type conform [CaseIterable](https://developer.apple.com/documentation/swift/caseiterable) and derive the values from there to prevent inconsistencies here down the line.